File: src/2022-intern/1290.convert-binary-number-in-a-linked-list-to-integer.js
/*
 * @lc app=leetcode id=1290 lang=javascript
 *
 * [1290] Convert Binary Number in a Linked List to Integer
 */
/**
 * @param {number} val
 * @param {ListNode|null} next
 */
function ListNode(val, next) {
  this.val = val;
  this.next = next;
}
// @lc code=start
/**
 * @param {ListNode} head
 * @return {number}
 */
const getDecimalValue = function (head) {
  let result = head.val;
  let currentNode = head.next;
  while (currentNode !== null) {
    result = (result << 1) + currentNode.val;
    currentNode = currentNode.next;
  }
  return result;
};
// @lc code=end

File: src/2025/1290.convert-binary-number-in-a-linked-list-to-integer.cpp
/*
 * @lc app=leetcode id=1290 lang=cpp
 *
 * [1290] Convert Binary Number in a Linked List to Integer
 */

struct ListNode {
  int val;
  ListNode* next;
  ListNode() : val(0), next(nullptr) {}
  ListNode(int x) : val(x), next(nullptr) {}
  ListNode(int x, ListNode* next) : val(x), next(next) {}
};

// @lc code=start
class Solution {
 public:
  int getDecimalValue(ListNode* head) {
    int result = 0;
    ListNode* current_node = head;
    while (current_node != nullptr) {
      result <<= 1;
      result += current_node->val;
      current_node = current_node->next;
    }
    return result;
  }
};
// @lc code=end
